src/modules/notifications/notifications.service.ts

```diff
@@ -35,6 +37,29 @@ export class NotificationsService {
     private userRepository: Repository<User>
   ) {}
 
+  async createGlobalNotifications(dto: CreateNotificationForAllUsersDto) {
+    const users = await this.userRepository.find();
+    try {
+      const notifications = users.map(user => {
+        return this.notificationRepository.create({
+          message: dto.message,
+          user,
+        });
+      });
+      await this.notificationRepository.save(notifications);
+      return {
+        status: 'success',
+        message: 'Notification created successfully',
+        data: null,
+      };
+    } catch (error) {
+      throw new CustomHttpException(
+        { message: 'An unexpected error occurred', error: error.message },
```

Comment:
just a string containing the error message and the status code is expected by the CustomHttpException handler.  reference auth for illustration

Reply:
Updated